apps/web/components/Gates.tsx

```diff
@@ -48,4 +53,38 @@ const Gates: React.FC<GateProps> = ({ children, gates, appData, dispatch }) => {
   return gateWrappers;
 };
 
+/**
+ * This BookerGates component is only used with the NEW booker.
+ * This component is responsible for fetching the bookers data
+ * (instead of accepting it as props) and then passing it along
+ * to the Gates component.
+ */
+export const BookerGates = ({ children }: { children: React.ReactNode }) => {
+  const event = useEvent();
+  const setEthSignature = useBookerStore((state) => state.setEthSignature);
+  const rainbowAppData = event?.data ? getEventTypeAppData(event.data, "rainbow") || {} : {};
+  const gates = [];
+  if (rainbowAppData && rainbowAppData.blockchainId && rainbowAppData.smartContractAddress) {
+    gates.push("rainbow");
+  }
+
+  const [gateState, gateDispatcher] = useReducer(
+    (state: GateState, newState: Partial<GateState>) => ({
+      ...state,
+      ...newState,
+    }),
+    {}
+  );
+
+  useEffect(() => {
+    setEthSignature(gateState.rainbowToken);
+  }, [gateState, setEthSignature]);
+
+  return (
+    <Gates gates={gates} appData={rainbowAppData} dispatch={gateDispatcher}>
+      {children}
+    </Gates>
+  );
+};
```

Comment:
does this always run, even if rainbow is disabled?

Reply:
Also we shouldn't load extra JavaScript for booking pages that don't need it